x-pack/plugins/security_solution/server/usage/collector.ts

```diff
@@ -308,6 +309,3368 @@ export const registerCollector: RegisterCollector = ({
               },
             },
           },
+          detection_rule_status: {
+            all_rules: {
+              eql: {
+                failed: {
+                  type: 'long',
+                  _meta: { description: 'The number of failed rules' },
+                },
+                top_failed: {
+                  '1': {
+                    message: {
+                      type: 'keyword',
+                      _meta: { description: 'Top 1 failed rule message' },
+                    },
+                  },
```

Comment:
@afharo 

I haven't done a visualization with arrays before. Is that well supported? I know that nested fields are still lacking support and is being tracked here:
https://github.com/elastic/kibana/issues/1084

So if I tried to group both the top "N" along side with the count of records as a nested field I don't think that is going to work with Kibana visualizations.

I will update the schema as I have a mistake but what I left out was the `count` ... It really should be this:

```sh
"1": {
  "properties": {
    "message": {
      "type": "keyword",
      "_meta": {
        "description": "Top 1 partial failed rule message"
      }
    },
    "count": {
      "type": "keyword",
      "_meta": {
        "description": "Top 1 partial failed rule message"
      }
    }
  }
}
```

If I changed that to an array with an object inside the array I don't know how easy that is going to be to visualize compared to just numbering them and stopping at 10.

Reply:
> Are nested types supported in this schema? I didn't see an example of it.

The schema is not a direct ES mapping (even thought it looks pretty similar 😅). You can specify the array of objects like this:

```suggestion
                top_failed: {
                  type: 'array',
                  items: {
                    message: {
                      type: 'keyword',
                      _meta: { description: 'One of top 10 failed rule messages' },
                    },
                    count: {
                      type: 'long',
                      _meta: { description: 'Number of times the message occurred' }
                    }
                  },
```

Then on the receiving end, we can work out different ways of storing it: for instance, I'd say that for easier consumption, we could loop through the array and index one document per object in the array with some common fields for correlation.

This way you get a "simple" event-like index to query and aggregate without depending on tree structures or nested types.

---------------------------------------------------------

As a separate comment, I was wondering if this type of data makes sense to send it as events with our Event-Based Telemetry client coming up on 8.3 (meta issue [here](https://github.com/elastic/kibana/issues/121992)).

@FrankHassanabad @pjhampton @peluja1012 wdyt? Is it worth the wait for one extra minor?

Reply:
Ok, I will change this to being an array with objects and update the code and tests with the array. Sounds good to me 👍 .

Reply:
@afharo Thank you for reviewing this PR and for your suggestion of taking advantage of the new Telemetry client. While I think it might make sense to use this new client in the future, we have been waiting for the Telemetry data this PR provides for a couple of releases. It's important that we start collecting it as soon as possible so we can get ahead of customer issues related to rule health and performance. I suggest we go with the approach presented in this PR in 8.2 and then migrate over to the Event-Based system in a future release after it's ready for consumption.

Reply:
@peluja1012 sounds like a good compromise to me. All clear! Thank you!